plugins/woocommerce/src/Internal/StockNotifications/Admin/SettingsController.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Internal\StockNotifications\Admin;
+
+/**
+ * Settings controller for Customer Stock Notifications.
+ */
+class SettingsController {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+
+		// Add a 'Customer stock notifications' section to Products settings.
+		add_filter( 'woocommerce_get_sections_products', array( $this, 'add_customer_stock_notifications_section' ), 100, 1 );
+
+		// Add the Customer Stock Notifications settings.
+		add_filter( 'woocommerce_get_settings_products', array( $this, 'add_customer_stock_notifications_settings' ), 100, 2 );
+
+		// Display admin notices about incompatible settings combinations.
+		add_action( 'admin_notices', array( $this, 'output_admin_notices' ) );
+	}
+
+	/**
+	 * Add a 'Customer stock notifications' section to Products settings.
+	 *
+	 * @param array $sections Products settings sections.
+	 * @return array New Products settings sections.
+	 */
+	public function add_customer_stock_notifications_section( $sections ) {
+		if ( ! is_array( $sections ) ) {
+			return $sections;
+		}
+
+		$section_title = __( 'Customer stock notifications', 'woocommerce' );
+
+		// Add 'Customer stock notifications' section to the Products tab, after Inventory.
+		$inventory_index = array_search( 'inventory', array_keys( $sections ), true );
+		if ( false !== $inventory_index ) {
+			$sections = array_slice( $sections, 0, $inventory_index + 1, true ) +
+				array( 'customer_stock_notifications' => $section_title ) +
+				array_slice( $sections, $inventory_index + 1, null, true );
+		} else {
+			$sections['customer_stock_notifications'] = $section_title;
+		}
+
+		return $sections;
+	}
+
+	/**
+	 * Add the Customer Stock Notifications settings.
+	 *
+	 * @param array  $settings Original settings.
+	 * @param string $section_id Settings section identifier.
+	 * @return array New settings.
+	 */
+	public function add_customer_stock_notifications_settings( $settings, $section_id ) {
+
+		if ( ! is_array( $settings ) ) {
+			return $settings;
+		}
+
+		if ( 'customer_stock_notifications' !== $section_id ) {
+			return $settings;
+		}
+
+		/**
+		 * Filter the Customer Stock Notifications settings.
+		 *
+		 * @since 0.0.0
+		 *
+		 * @param array $default_customer_stock_notifications_settings The default Customer Stock Notifications settings.
+		 */
+		$stock_notification_settings = apply_filters(
+			'woocommerce_customer_stock_notifications_settings',
```

Comment:
I believe we can omit this hook. Anyone can hook before or after the `woocommerce_get_settings_products` to achieve the same results.

Reply:
Makes sense! Thanks for elaborating!